pbj-core/pbj-grpc-helidon/src/main/java/com/hedera/pbj/grpc/helidon/PbjProtocolSelector.java

```diff
@@ -0,0 +1,160 @@
+
+/**
+ * Sub-protocol selector for HTTP/2. This is the main entry point into the PBJ implementation of gRPC. The web
+ * server will use this class to determine if a request is a gRPC request and if so, how to handle it.
+ */
+class PbjProtocolSelector implements Http2SubProtocolSelector {
+    /** This routing has absolutely no routes. */
+    private static final PbjRouting EMPTY = PbjRouting.builder().build();
+
+    private final PbjConfig config;
+    private final DeadlineDetector deadlineDetector;
+    private final ScheduledExecutorService deadlineExecutorService = Executors.newSingleThreadScheduledExecutor();
+    private final Counter requestCounter;
+    private final Counter failedRequestCounter;
+
+    /**
+     * Create a new PBJ based grpc protocol selector (default). Access restricted to be package-private so as
+     * to limit instantiation to the {@link PbjProtocolProvider}.
+     */
+    PbjProtocolSelector(@NonNull final PbjConfig config) {
+        this.config = requireNonNull(config);
+        this.deadlineDetector = (deadline, onDeadlineExceeded) ->
+                deadlineExecutorService.schedule(onDeadlineExceeded, deadline, TimeUnit.NANOSECONDS);
+
+        final var metricRegistry = Metrics.globalRegistry();
+        this.requestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.requests")
+                .scope("vendor")
+                .description("The number of gRPC requests"));
+        this.failedRequestCounter = metricRegistry.getOrCreate(Counter.builder("pbj.grpc.request.failures")
+                .scope("vendor")
+                .description("The number of failed gRPC requests"));
+    }
+
+    /**
+     * Called by Helidon to create the sub-protocol for PBJ gRPC requests. The {@link SubProtocolResult} returned
+     * will be responsible for handling the request.
+     */
+    @Override
+    public SubProtocolResult subProtocol(ConnectionContext ctx,
+                                         HttpPrologue prologue,
+                                         Http2Headers headers,
+                                         Http2StreamWriter streamWriter,
+                                         int streamId,
+                                         Http2Settings serverSettings, // unused
+                                         Http2Settings clientSettings, // unused
+                                         StreamFlowControl flowControl,
+                                         Http2StreamState currentStreamState,
+                                         Router router) {
+        this.requestCounter.increment();
+
+        // As per the specification, only POST requests are supported. I would have thought that the code here should
+        // return a response code of 405 (Method Not Allowed) if the method is not POST, but the code here just returns
+        // NOT_SUPPORTED. I'm not sure if this is technically correct.
+        if (prologue.method() != Method.POST) {
```

Comment:
This code is owned by us though and we can do what we want (in fact we could support GET, I think), so I'm unsure about the reasoning. If this particular behavior of "UNSUPPORTED" vs 405 is specified in the Helidon spec somewhere, please add a link to that doc here in code for reference.

Reply:
I've not managed to find any helidon docs specifying this, @rbair23 do you have a source?

Reply:
Might be to meet the requirements of the [gRPC specification](https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md)

Reply:
This does appear to be the case. Adding a link in the comment.